core/src/main/scala/org/apache/spark/api/r/RRDD.scala

```diff
@@ -127,6 +127,13 @@ private[r] object RRDD {
       sparkConf.setExecutorEnv(name.toString, value.toString)
     }
 
+    if (sparkEnvirMap.containsKey("spark.r.sql.default.derby.dir") &&
```

Comment:
well, in revisiting this I thought it would be easier to minimize the impact by making this R only.
it would be much easier if we make the derby log going to tmp always for all lang binding